File: app/src/main/java/com/example/grocer21/Database/AppDatabase.java
package com.example.grocer21.Database;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import android.os.AsyncTask;
import android.support.annotation.NonNull;

@Database(entities = {Food.class, Allergy.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    public abstract FoodDao foodDao();
    public abstract AllergyDao allergyDao();

    private static volatile AppDatabase INSTANCE;

    public static AppDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "app_database").addCallback(appDatabaseCallback)
                            .build();
                }
            }
        }
        return INSTANCE;
    }

    private static RoomDatabase.Callback appDatabaseCallback = new RoomDatabase.Callback(){

                @Override
                public void onOpen (@NonNull SupportSQLiteDatabase db) {
                    super.onOpen(db);
                    new PopulateDbAsync(INSTANCE).execute();
                }
            };

    private static class PopulateDbAsync extends AsyncTask<Void, Void, Void> {

        private final FoodDao fDao;
        private final AllergyDao aDao;

        PopulateDbAsync(AppDatabase db) {
            fDao = db.foodDao();
            aDao = db.allergyDao();
        }

        @Override
        protected Void doInBackground(final Void... params) {
            fDao.deleteAll();
            Food food = new Food("Chicken", 1111111111);
            fDao.insert(food);
            food = new Food("Pork", 1111111112);
            fDao.insert(food);

            aDao.deleteAll();
            Allergy allergy = new Allergy("Gluten");
            aDao.insert(allergy);
            allergy = new Allergy("Shellfish");
            aDao.insert(allergy);
            return null;
        }
    }
}

File: app/src/main/java/com/example/grocer21/database/AppDatabase.kt
package com.example.grocer21.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.grocer21.database.daos.*
import com.example.grocer21.database.entities.*

/**
 * class which represents the database
 *
 * [Room] implementation
 */
@Database(entities = [
    Allergies::class,
    DietFoods::class,
    Diets::class,
    FoodAllergens::class,
    Foods::class,
    Profile::class,
    ShoppingList::class,
    UserAllergies::class,
    UserDiets::class
], version = 1)
abstract class AppDatabase : RoomDatabase() {
    /**
     * Constructor definition method for the [allergiesDao] interface
     */
    abstract fun allergiesDao(): AllergiesDao

    /**
     * Constructor definition method for the [dietFoodsDao] interface
     */
    abstract fun dietFoodsDao(): DietFoodsDao

    /**
     * Constructor definition method for the [dietsDao] interface
     */
    abstract fun dietsDao(): DietsDao

    /**
     * Constructor definition method for the [foodAllergensDao] interface
     */
    abstract fun foodAllergensDao(): FoodAllergensDao

    /**
     * Constructor definition method for the [foodsDao] interface
     */
    abstract fun foodsDao(): FoodsDao

    /**
     * Constructor definition method for the [profileDao] interface
     */
    abstract fun profileDao(): ProfileDao

    /**
     * Constructor definition method for the [shoppingListDao] interface
     */
    abstract fun shoppingListDao(): ShoppingListDao

    /**
     * Constructor definition method for the [userAllergiesDao] interface
     */
    abstract fun userAllergiesDao(): UserAllergiesDao

    /**
     * Constructor definition method for the [userDietsDao] interface
     */
    abstract fun userDietsDao(): UserDietsDao

    companion object {
        @Volatile
        internal var instance: AppDatabase? = null

        /**
         * function to fetch the [AppDatabase]'s current iteration
         */
        fun getDatabase(context: Context): AppDatabase {
            val tempInstance = instance
            if (tempInstance != null) {
                return tempInstance
            }
            synchronized(this) {
                val mInstance = Room.databaseBuilder(
                        context.applicationContext,
                        AppDatabase::class.java,
                        "App_database"
                ).build()
                instance = mInstance
                return mInstance
            }
        }
    }
}
